### PR TITLE
net: optimize restore_rule() to avoid opening the CR_FD_RULE image file twice

### PR DESCRIPTION
**Pull request by nviennot**
Previously, `open_image(CR_FD_RULE, O_RSTR, pid)` was called twice.
Opening an image file twice is not allowed when streaming the image.
This commit optimizes the code to only open the image file once.

Also improved the error path in restore_ip_dump().

### CODE REVIEW

**Comment by rst0git** [comment]
I can confirm that this would fix the error with `restore_rule()`, however, the same problem exists with `restore_route()`. It calls `restore_ip_dump()` twice.

File: criu/net.c [hunk 1]
```diff
@@ -2264,25 +2264,34 @@ static int restore_ip_dump(int type, int pid, char *cmd)
...
 
-close:
+	ret = run_ip_tool(cmd, "restore", NULL, NULL, fileno(tmp_file), -1, 0);
+
+out2:
 	if (fclose(tmp_file)) {
 		pr_perror("Failed to close tmpfile");
 	}
 
+out1:
```
[the remark this answers is not in the document]
Comment (rst0git) [reply criu/net.c:2292]: +1
It improves readability.

**Review by rst0git — APPROVED** [review]
LGMT

File: criu/net.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (nviennot) [reply criu/net.c]: Great, thank you

**Review by rst0git — APPROVED** [review]

**Pull request merged**